### PR TITLE
Relax calculations for CG electronic minimizations

### PR DESCRIPTION
**Pull request by degennar**
The relax procedure was switched off whenever the electronic degrees of freedom where minimized via the conjugate gradient technique.

`relax` calculations can now be performed with KCP at DFT, PZ and Koopmans levels. I also run the following tests [FINISHED]:
- DFT relaxation gives the same results of a PWrelax calculation (within 10^-4 bohr for N_2)
- We had already checked that the PZ forces (computed with Hellmann-Feynman) are consistent with the derivatives of the energy wrt the ionic displacements. The `relax` calculation with PZ is using the PZ forces.
- At the Koopmans level the KI forces (and relaxation) are actually consistent with the DFT forces, while KIPZ forces are consistent with (PZ forces)

Also, when running the code now, after the first electronic minimization, the code was doing a couple of additional electronic minimizations (not superclear why!). To avoid this is sufficient to set `nstep=1`. I did not want to set `nstep=1` by default (in general) so the code will avoid these additional steps for standard KCP calculations (i.e. when `calculation='cp'`) and when `ion_dynamics='none'`.

`vc-relax` calculations cannot be done for the moment, since the stress tensor is not well computed (there is an open issue associated to this).

**NB**: the time-step (gauging the size of the ionic displacement upon the application of the forces) is defined by the input variable `dt`, which by default is set to 1 (too small). In order to speed-up the relaxation one should set it at least at 10. So the parameters that one should set when running a `relax` calculation are:
```
&SYSTEM
  calculation = 'relax'
  nstep = 100 (e.g.)
  dt = 50 (e.g.)
&IONS
  ion_dynamics = 'damp'
  ion_damping = 0.5 (e.g.)
```

@elinscott Since this is a small PR we may merge this before #13 .

### CODE REVIEW

File: CPV/cpr.f90 [hunk 1]
```diff
@@ -956,8 +953,6 @@ SUBROUTINE cprmain(tau_out, fion_out, etot_out)
       !
       delta_etot = ABS(epre - enow)
       !
-      tstop = check_stop_now() .OR. tlast
-      !
```
Comment (degennar) [inline CPV/cpr.f90:960]: @nscolonna non mi sembra di alcuna utilita' questa call o sbaglio? L'ho levata perche' nel controllare i restart del loop mi sembra che crei solo confusione.

Comment (nscolonna) [reply CPV/cpr.f90]: It's also re-defined few line below (962 old 957 new). Ok to remove

**Comment by degennar** [comment]
OK also the tests on `koopmans` are passing, so @nscolonna you can review whenever you are ready. Thanks!

File: CPV/input.f90 [hunk 2]
```diff
@@ -536,6 +536,13 @@ SUBROUTINE set_control_flags()
          !
       END SELECT
       !
+      ! ... For standard KCP calculations we set nstep to 1 by default
+      !
+      IF (calculation == 'cp' .and. ion_dynamics == 'none') THEN
+        nomore_ = 1
+        nstep_ = 1
+      ENDIF
+      !
```
Comment (degennar) [inline CPV/input.f90:545]: Needs further check. It probably prevents any step in a damp/sd electron dynamics.

**Comment by nscolonna** [comment]
Tested on H2 dissociation. Attached scripts and results for reference  
[H2_pes.tar.gz](https://github.com/user-attachments/files/16965059/H2_pes.tar.gz)
[H2_PES_zoom.pdf](https://github.com/user-attachments/files/16965061/H2_PES_zoom.pdf)
[H2_PES.pdf](https://github.com/user-attachments/files/16965062/H2_PES.pdf)

**Review by nscolonna — APPROVED** [review]
Thanks! and sorry it took so long... 
See my latest comment for the tests I did.

**Pull request merged**